app-typescript/components/TabList.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface ITabList {
+    size?: 'normal' | 'big' | 'small';
+    theme?: 'light' | 'dark';
+    children: Array<any>;
+}
+
+interface IState {
+    index: number;
+}
+
+interface ITab {
+    text: string;
+}
+
+class Tab extends React.PureComponent<ITab> {
+    render() {
+        return (
+            <React.Fragment>
+                {this.props.children}
+            </React.Fragment>
+        );
+    }
+}
+
+class TabList extends React.PureComponent<ITabList, IState> {
+    constructor(props: ITabList) {
+        super(props);
+        this.state = {
+            index: 0,
+        };
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(index: number) {
```

Comment:
Can you make this `private` and add a `public` method `goTo(label: string)` which would enable a tab given a label. It would be called by using a ref.